NewForce-at-Mountwest/bangazon-site-take-2-centaurs
# Authenticated User Can Access Account Profile in Navigation Bar, 'Click' the Hyperlink for "Payment Options", Add (Populate), Edit, View Details, and Delete [their] User Payment Option(s)

# Description

## Add Payment Option
Given the User is Authenticated
When the User Performs a Gesture on the "[user e-mail] Profile" Affordance in the Navigation Bar
And the User 'Clicks' on the "Payment Options" Hyperlink in their Profile (left-hand side of page)
Then the User can Select "Add a Payment Option" to Populate the "Create New Payment Type(s)" Form
And 'Click' Submit to Save Payment Option

## Delete Payment Option
Given the User is Authenticated
When the User Performs a Gesture on the "[user e-mail] Profile" Affordance in the Navigation Bar
And the User 'Clicks' on the "Payment Options" Hyperlink in their Profile (left-hand side of page)
Then the User can Select the "Delete" Hyperlink Next to the Targeted Payment Type Planned for Erasure
And the User is then Prompted to Confirm / Selects Deletion of the Aforementioned Payment Option ("Delete" Button to Confirm)
Then the Payment Option will get Deleted

## Edit Payment Option
Given the User is Authenticated
When the User Performs a Gesture on the "[user e-mail] Profile" Affordance in the Navigation Bar
And the User 'Clicks' on the "Payment Options" Hyperlink in their Profile (left-hand side of page)
Then the User can Select the "Edit" Hyperlink Next to the Targeted Payment Type Planned for Altering
And the User can View / Edit this Payment Option ("Save" Button to Submit)
Then the Payment Option will be Edited

## View Payment Option Details
Given the User is Authenticated
When the User Performs a Gesture on the "[user e-mail] Profile" Affordance in the Navigation Bar
And the User 'Clicks' on the "Payment Options" Hyperlink in their Profile (left-hand side of page)
Then the User can Select the "Details" Hyperlink Next to the Targeted Payment Type to be Viewed
And the User can View this Payment Option
And has an Option for Editing the Payment Option (via "Edit" Hyperlink)
Then the Payment Option will be Viewed / Edited / Confirmed


### Related Ticket(s)
1. Trello Ticket No. 7: Users can Add Payment Type (https://trello.com/c/syBkfxhc)
1. Trello Ticket No. 8: Users can Delete Payment Type (https://trello.com/c/ORnb1FPh)


### Steps to Test
1. "git add ." and "git commit [-m]" whatever you're working on (on your branch)
2. "git checkout master", to switch to our local master branch
3. "git fetch --all" will give you access to the branch in question
4. "git checkout bang-jw" to switch to this branch (PR review requested)
5. If you do not see the most recent version of branch files, try "git pull origin bang-jw"
1. In Git Bash, 'cd' to "bangazon-site-take-2-centaurs"
1. Start "BangazonSite.sln"
1. Go to Packet Manager Console and insert command: "add-migration" (changes made)
1. Go to Packet Manager Console and insert command: "update-database" (changes made)
1. Press F5 to run browser / CTRL+R, A to run tests
1. In browser, 'click' "Login" to reach the sign in page
1. Login as Email: "admin@admin.com"; Password: "Admin8*"
1. In browser, 'click' "admin@admin.com Profile" to reach the profile page
1. Under the management account settings tab (left-hand side of screen) 'click' the "Payment Options" Hyperlink
1. User should see a link to "Add a Payment Option" and a list of their submitted Payment Options, including: "DateCreated", "Description", and "AccountNumber" information.
1. Click "Add a Payment Option" to access / populate "Create New Payment Type(s)" form fields and hit "Submit" Button to save OR 'click the "Back to List" link to return to the list of payment types
1. In Payment Option Index ("Profile Payment Options") user can 'click' on "Edit", "Details", and/or "Delete" link options - to the right of their targeted payment type
1. Selecting the "Details" hyperlink, next to the targeted payment type to be viewed, allows the user to 
view this payment option and offers an option for editing this payment option (via an "Edit" hyperlink)
1. Selecting the aforementioned "Edit" hyperlink (or the one next to the targeted payment type, planned for alteration on the list view) allows user to view / edit this payment option with a "Save" button affordance to submit
1. The user can also select the "Delete" hyperlink beside [a] targeted payment type planned for erasure, confirm the deletion via a warning prompt ("delete" button to confirm), and remove payment type(s)


# Checklist:
